geth/node/utils.go

```diff
@@ -10,16 +10,16 @@ import (
 // HaltOnPanic recovers from panic, logs issue, sends upward notification, and exits
 func HaltOnPanic() {
 	if r := recover(); r != nil {
-		err := fmt.Errorf("%v: %v", ErrNodeRunFailure, r)
+		strErr := fmt.Sprintf("%v: %v", ErrNodeRunFailure, r)
 
 		// send signal up to native app
 		signal.Send(signal.Envelope{
 			Type: signal.EventNodeCrashed,
 			Event: signal.NodeCrashEvent{
-				Error: err.Error(),
+				Error: strErr,
 			},
 		})
 
-		common.Fatalf(err) // os.exit(1) is called internally
+		common.Fatalf(ErrNodeRunFailure, r) // os.exit(1) is called internally
```

Comment:
Shouldn't it be `common.Fatalf(strErr)` in order not to change the logic? I doubt this is tested.

Reply:
I'd if we can use logrus here which is better option than `fmt.Fprintf` for writing errors to STDERR we'd have more stable and predictable code.

Meanwhile, it works okay: https://play.golang.org/p/z8YP9PMT8A0